### ISSUE TITLE
install_github always installs even if no change

### ISSUE THREAD

**Issue by trinker** [r-lib/remotes#122]
When `install_github` was part of the devtools package it would check github and only install if the SHA (I'm going from memory but something than SHA) there was different from what was currently installed.  This meant I could have 

```
install_github('foo/bar')
library('bar')
```

in an Rmd and bar was only installed if there was a change.  Now if you have several GitHub packages it slows the process down b/c the packages are installed no matter what even when there is no change.  Could the old behavior be reinstated?

**Comment by trinker** [comment]
Checking in on this.  We used `remotes::install_github` in the pacman package in May 2017 and at the time it did this checking.  It now is at the top of code everywhere...Including .Rmd files.  This week I revisited a data sci. project where a report was created from a template for many departments.  Each loop through uses the .Rmd with the call to `remotes::install_github` which installs multiple packages each and every iteration.

**Comment by gaborcsardi** [comment]
Yes, it does that, indeed.

**Comment by trinker** [comment]
@gaborcsardi Sorry I don't follow.  Do you mean that remotes checks github and if the current local version is the same as the github version the new version is not installed?

If so then maybe this is a problem with my R version or version of remotes or something else.  Below I show a MWE and the output I see if I run the line `remotes::install_github('trinker/pysty')` twice.  You can see that both times this line is run the package is downloaded and installed.

## Code

```
if (!require("remotes")) install.packages("remotes")
remotes::install_github('trinker/pysty')
remotes::install_github('trinker/pysty')
```

## Output

```
> if (!require("remotes")) install.packages("remotes")
Loading required package: remotes

## Run 1
> remotes::install_github('trinker/pysty')
Downloading GitHub repo trinker/pysty@master
* installing *source* package 'pysty' ...
** R
** inst
** preparing package for lazy loading
** help
*** installing help indices
  converting help for package 'pysty'
    finding HTML links ... done
    add_imports                             html  
    clear_imports                           html  
    colons                                  html  
Rd warning: C:/Users/trinker/AppData/Local/Temp/RtmpaqkHd8/remotes1170503313ce/trinker-pysty-59713d5/man/colons.Rd:29: missing file link '::'
    get_imports                             html  
    get_imports_functions                   html  
    pysty                                   html  
** building package indices
** testing if installed package can be loaded
* DONE (pysty)
In R CMD INSTALL

## Run 2
> remotes::install_github('trinker/pysty')
Downloading GitHub repo trinker/pysty@master
* installing *source* package 'pysty' ...
** R
** inst
** preparing package for lazy loading
** help
*** installing help indices
  converting help for package 'pysty'
    finding HTML links ... done
    add_imports                             html  
    clear_imports                           html  
    colons                                  html  
Rd warning: C:/Users/trinker/AppData/Local/Temp/RtmpaqkHd8/remotes117059d3e9b/trinker-pysty-59713d5/man/colons.Rd:29: missing file link '::'
    get_imports                             html  
    get_imports_functions                   html  
    pysty                                   html  
** building package indices
** testing if installed package can be loaded
* DONE (pysty)
In R CMD INSTALL
```

## Session Info

```
> sessionInfo()
R version 3.4.3 (2017-11-30)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 16299)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] remotes_1.1.1

loaded via a namespace (and not attached):
[1] compiler_3.4.3 tools_3.4.3   
```

This is the way devtools used to handle this:

```
> devtools::install_github('trinker/pysty')
Downloading GitHub repo trinker/pysty@master
from URL https://api.github.com/repos/trinker/pysty/zipball/master
Installing pysty
"C:/R/R-3.3.3/bin/x64/R" --no-site-file --no-environ --no-save --no-restore  \
  --quiet CMD INSTALL  \
  "C:/Users/trinker/AppData/Local/Temp/Rtmp4aHjXK/devtools1e447bf666af/trinker-pysty-59713d5"  \
  --library="C:/R/R-3.3.3/library" --install-tests 

* installing *source* package 'pysty' ...
** R
** inst
** tests
** preparing package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (pysty)


> devtools::install_github('trinker/pysty')
Skipping install of 'pysty' from a github remote, the SHA1 (59713d57) has not changed since last install.
  Use `force = TRUE` to force installation
```

**Comment by jimhester** [comment]
remotes was split from devtools before https://github.com/r-lib/devtools/pull/903, which adds this functionality to devtools was merged. Therefore remotes has never had this functionality.

**Comment by trinker** [comment]
@jimhester Gotcha.  Thanks.  It sounds like this is an intentional design choice then.

Is this a possibility as a feature or do I need to find some other way forward?  
If the answer is no may I ask why?

**Comment by gaborcsardi** [comment]
It is not a design choice. PRs are welcome. :)

**Comment by dracodoc** [comment]
@trinker You can use `devtools` temporarily (that may not be a optimal solution because it has heavier dependencies). I think `devtools` has the plan to delegate related functionality to `remotes`, but there are quite some differences in both branch so it'll be a difficult task.

**Comment by jimhester** [comment]
This is fixed by #135
